Source/VMware.vSphereDSC/DSCResources/DistributedSwitch.ps1

```diff
@@ -0,0 +1,363 @@
+
+        if ($this.LinkDiscoveryProtocol -ne [LinkDiscoveryProtocolProtocol]::Unset) {
+            $shouldUpdateDistributedSwitch += ($this.LinkDiscoveryProtocol.ToString() -ne $distributedSwitch.LinkDiscoveryProtocol.ToString())
+        }
+
+        if ($this.LinkDiscoveryProtocolOperation -ne [LinkDiscoveryProtocolOperation]::Unset) {
+            $shouldUpdateDistributedSwitch += ($this.LinkDiscoveryProtocolOperation.ToString() -ne $distributedSwitch.LinkDiscoveryProtocolOperation.ToString())
+        }
+
+        return ($shouldUpdateDistributedSwitch -Contains $true)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns the populated Distributed Switch parameters.
+    #>
+    [hashtable] GetDistributedSwitchParams() {
+        $distributedSwitchParams = @{}
+
+        $distributedSwitchParams.Server = $this.Connection
+        $distributedSwitchParams.RunAsync = $true
+        $distributedSwitchParams.Confirm = $false
+        $distributedSwitchParams.ErrorAction = 'Stop'
+
+        if (![string]::IsNullOrEmpty($this.ContactDetails)) { $distributedSwitchParams.ContactDetails = $this.ContactDetails }
+        if (![string]::IsNullOrEmpty($this.ContactName)) { $distributedSwitchParams.ContactName = $this.ContactName }
+        if (![string]::IsNullOrEmpty($this.Notes)) { $distributedSwitchParams.Notes = $this.Notes }
+        if (![string]::IsNullOrEmpty($this.Version)) { $distributedSwitchParams.Version = $this.Version }
+
+        if ($null -ne $this.MaxPorts) { $distributedSwitchParams.MaxPorts = $this.MaxPorts }
+        if ($null -ne $this.Mtu) { $distributedSwitchParams.Mtu = $this.Mtu }
+        if ($null -ne $this.NumUplinkPorts) { $distributedSwitchParams.NumUplinkPorts = $this.NumUplinkPorts }
+
+        if ($this.LinkDiscoveryProtocol -ne [LinkDiscoveryProtocolProtocol]::Unset) {
+            $distributedSwitchParams.LinkDiscoveryProtocol = $this.LinkDiscoveryProtocol.ToString()
+        }
+
+        if ($this.LinkDiscoveryProtocolOperation -ne [LinkDiscoveryProtocolOperation]::Unset) {
+            $distributedSwitchParams.LinkDiscoveryProtocolOperation = $this.LinkDiscoveryProtocolOperation.ToString()
+        }
+
+        return $distributedSwitchParams
+    }
+
+    <#
+    .DESCRIPTION
+
+    Creates a new Distributed Switch with the specified properties at the specified location.
+    #>
+    [void] AddDistributedSwitch($distributedSwitchLocation) {
+        $distributedSwitchParams = $this.GetDistributedSwitchParams()
+        $distributedSwitchParams.Name = $this.Name
+        $distributedSwitchParams.Location = $distributedSwitchLocation
+
+        <#
+        ReferenceVDSwitch and WithoutPortGroups are parameters only for the New-VDSwitch cmdlet
+        and are not used for the Set-VDSwitch cmdlet.
+        #>
+        if (![string]::IsNullOrEmpty($this.ReferenceVDSwitch)) { $distributedSwitchParams.ReferenceVDSwitch = $this.ReferenceVDSwitch }
```

Comment:
ReferenceVDSwitch will trigger OBN of the cmdlet parameter. This would be problem if multiple vdswitches are resolved

Reply:
The VDSwitch Name is unique for each vCenter, so triggering the OBN would not be a problem because only one Distributed Switch will be resolved.